src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/DebugProfileDebugTargetGenerator.cs

```diff
@@ -15,101 +16,49 @@ namespace Microsoft.VisualStudio.ProjectSystem.Debug
     [AppliesTo(ProjectCapability.LaunchProfiles)]
     [Export(typeof(IDynamicDebugTargetsGenerator))]
     [ExportMetadata("Name", "DebugProfileProvider")]
-    internal class DebugProfileDebugTargetGenerator : ProjectValueDataSourceBase<IReadOnlyList<IEnumValue>>, IDynamicEnumValuesProvider, IDynamicDebugTargetsGenerator
```

Comment:
This component was changed from `ProjectValueDataSourceBase<>` to `ChainedProjectValueDataSourceBase<>`.

The latter is a simple alternative when you are producing data that's entirely derived from other data sources.

Reply:
What are the data sources this generator is getting the data from? Is one source the `IVersionedLaunchSettingsProvider`?

Reply:
`ChainedProjectValueDataSourceBase` has one abstract method you have to implement where you link upstream data source(s) to the target block you are passed. You can see the sources in that method.